### DISCUSSION TITLE
Beach access

### DISCUSSION THREAD

**Discussion by cpenny86** [loathers/garbage-collector#1716] [Ideas]
Running garbo today after a CS run, it got stuck in a loop since I had forgotten to get beach access. Easy fix once I noticed it, but would it be worth telling the code to just buy a bus pass from the general store if it can't otherwise find the beach?

**Comment by wRAR** [comment]
(meatcar, not bus pass, as Mafia won't buy a bus pass)

**Comment by Rinn** [comment]
it looks like we have a function to make the meatcar if the desert is locked. afaik I've been running a lot of CS loops this year and haven't noticed any issues (and may be less needed now that we can't use the shore for resetting the orb) https://github.com/loathers/garbage-collector/blob/b0664d407871dbca13f31fd05ce29759ae4abdc6/packages/garbo/src/fights.ts#L2246

**Comment by wRAR** [comment]
The usual way users encounter the beach access infinite loop is attempting to fight a sandworm. As for `ensureBeachAccess()`, it looks like a no-op to me, at least after CS (maybe there are account states where it's useful), for the same reason as Mafia refusing to buy a bus pass: [this old well-known bug report](https://kolmafia.us/threads/lastdesertunlock-improperly-set.26371/).
